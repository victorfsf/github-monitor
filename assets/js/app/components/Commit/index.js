import Commit from './Commit';

export default Commit;
