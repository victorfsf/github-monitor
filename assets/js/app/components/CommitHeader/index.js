import CommitHeader from './CommitHeader';

export default CommitHeader;
