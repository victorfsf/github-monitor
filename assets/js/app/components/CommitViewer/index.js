import CommitViewer from './CommitViewer';

export default CommitViewer;
