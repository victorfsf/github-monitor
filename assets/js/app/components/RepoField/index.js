import RepoField from './RepoField';

export default RepoField;
