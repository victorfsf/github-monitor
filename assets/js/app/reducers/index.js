import { combineReducers } from 'redux';

// TODO: write actual reducers
const mockedReducer = (state, action) => ({ mock: action });

const rootReducer = combineReducers({
  mockedReducer,
});

export default rootReducer;
