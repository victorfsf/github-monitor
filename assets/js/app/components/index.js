import Box from './Box';
import CommitViewer from './CommitViewer';
import Container from './Container';
import LoginForm from './LoginForm';
import RepoField from './RepoField';
import Spinner from './Spinner';
import Pagination from './Pagination';
import NotFound from './NotFound';

export {
  Box,
  CommitViewer,
  Container,
  LoginForm,
  RepoField,
  Spinner,
  Pagination,
  NotFound,
};
