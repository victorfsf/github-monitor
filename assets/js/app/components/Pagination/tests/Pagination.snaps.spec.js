import React from 'react';
import renderer from 'react-test-renderer';
import { Pagination } from 'app/components';


describe('Pagination', () => {
  let Component;
  let tree;

  test('does not render without functions', () => {
    global.console = {
      error: jest.fn(),
    };
    renderer.create(<Pagination />);
    renderer.create(<Pagination onClickNext={e => e} />);
    renderer.create(<Pagination onClickPrev={e => e} />);
    expect(global.console.error).toHaveBeenCalledTimes(2);
  });

  test('renders', () => {
    Component = renderer.create(
      <Pagination onClickNext={e => e} onClickPrev={e => e} next />,
    );
    tree = Component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
