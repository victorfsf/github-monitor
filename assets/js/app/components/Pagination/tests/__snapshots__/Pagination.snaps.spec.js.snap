// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination renders 1`] = `
<nav
  aria-label="Commits navigation"
  className="mt-4 mb-5"
>
  <ul
    className="pagination justify-content-center"
  >
    <li
      className="page-item "
    >
      <button
        className="page-link"
        onClick={[Function]}
        type="button"
      >
        Newer
      </button>
    </li>
    <li
      className="page-item disabled"
    >
      <button
        className="page-link"
        onClick={[Function]}
        type="button"
      >
        Older
      </button>
    </li>
  </ul>
</nav>
`;
